urlc8dbc7ce6e_dlubarov_guava_tgz-pJ8-MainJ8/src/d/Opcodes.java

```diff
@@ -0,0 +1,251 @@
+package d;
+
+import java.util.Arrays;
+
+import common.*;
+
+import d.ty.desc.TypeDesc;
+
+public final class Opcodes {
+    public static final int
+            POP = 0,
+            DUP = 1,
+            // Literals
+            CONST_TRUE = 10,
+            CONST_FALSE = 11,
+            CONST_INT = 12,
+            CONST_CHAR = 13,
+            CONST_LONG = 14,
+            CONST_DOUBLE = 15,
+            CONST_STRING = 16,
+            CREATE_SEQ = 18,
+            // Locals and fields
+            GET_LOCAL = 20,
+            PUT_LOCAL = 21,
+            GET_STATIC_FIELD = 22,
+            PUT_STATIC_FIELD = 23,
+            GET_INSTANCE_FIELD = 24,
+            PUT_INSTANCE_FIELD = 25,
+            // Methods
+            INVOKE_STATIC = 30,
+            INVOKE_VIRTUAL = 31,
+            NEW = 40,
+            NEW_NO_GENERICS = 41,
+            // Control
+            JUMP = 50,
+            JUMP_COND = 51,
+            RETURN = 55,
+            RETURN_UNIT = 56,
+            // Common operations
+            BOOL_NEG = 60,
+            INT_NEG = 100,
+            INT_ADD = 110,
+            INT_SUB = 111,
+            INT_MUL = 112,
+            INT_DIV = 113,
+            INT_MOD = 114,
+            INT_IOR = 115,
+            INT_XOR = 116,
+            INT_AND = 117,
+            INT_LSHIFT = 118,
+            INT_RSHIFT_UNSIGNED = 119,
+            INT_RSHIFT_SIGNED = 120,
+            INT_COMPARETO = 121,
+            INT_LT = 122,
+            INT_LTE = 123,
+            INT_GT = 124,
+            INT_GTE = 125;
+
+    private Opcodes() {}
+
+    public static String repr(int[] code,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 124 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.